### PR TITLE
chore(datadog_agent source): add an integration test

### PR DESCRIPTION
**Pull request by jdrouet**
There is currently no test on the `datadog-agent` source.
In the current PR, we start the agent in a docker container. The test runner then sends a logs to the agent that should forward it back to vector.

Unsure part: using `network_mode: host` might not work on the CI but is required for the agent to send some data to vector.

Signed-off-by: Jérémie Drouet <jeremie.drouet@datadoghq.com>

### CODE REVIEW

File: scripts/setup_integration/datadog-agent.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+set -o pipefail
+
+# redis_integration_env.sh
```
Comment (jszwedko) [inline scripts/setup_integration/datadog-agent.sh:4]:
```suggestion
# datadog_agent_env.sh
```

The filename here should be updated too.

File: scripts/setup_integration/datadog-agent.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml down
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml rm -f
+}
+
+stop_docker () {
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml down
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml rm -f
+}
+
+echo "Running $ACTION action for Redis integration tests environment"
```
Comment (jszwedko) [inline scripts/setup_integration/datadog-agent.sh:39]:
```suggestion
echo "Running $ACTION action for Datadog Agent integration tests environment"
```

File: scripts/setup_integration/datadog-agent.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+stop_podman () {
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml down
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml rm -f
+}
+
+stop_docker () {
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml down
+  docker-compose -f scripts/setup_integration/docker-compose.datadog-agent.yml rm -f
+}
```
Comment (jszwedko) [inline scripts/setup_integration/datadog-agent.sh:37]: These look the same to me between docker and podman. Do we need both sets?

**Review by jszwedko — APPROVED** [review]
Great! I'm happy to see this. I left a couple of small comments, but overall looks good.

File: scripts/setup_integration/datadog-agent.sh (hunk 3 above)
Comment (jdrouet) [reply scripts/setup_integration/datadog-agent.sh:37]: Yeah, I did a dumb copy paste 😓 Will fix that.

**Comment by jdrouet** [comment]
seems like the tests are not executed

File: src/sources/datadog/tests.rs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+        async move {
+            let mut stream = TcpStream::connect("0.0.0.0:8181").unwrap();
+            let data = "hello world\nit's vector speaking\n";
+            stream.write_all(data.as_bytes()).unwrap();
+        },
+        recv,
+        2,
+    )
+    .await;
+    assert!(!events.is_empty());
```
Comment (prognant) [inline src/sources/datadog/tests.rs:29]: We probably should check at least the `message` fields.

**Review by prognant — APPROVED** [review]
Left minor comment but overall LGTM.

File: scripts/setup_integration/docker-compose.datadog-agent.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    # we need to enable the network mode considering datadog-agent needs to
+    # communicate with vector running on the host.
+    network_mode: host
+    environment:
+      - DD_API_KEY=${CI_TEST_DATADOG_API_KEY}
+      - DD_LOGS_ENABLED=true
+      - DD_LOGS_CONFIG_LOGS_DD_URL=0.0.0.0:8080
+      - DD_LOGS_CONFIG_LOGS_NO_SSL=true
+      - DD_LOGS_CONFIG_USE_HTTP=true
+      - DD_LOGS_CONFIG_USE_V2_API=false
```
[the remark this answers is not in the document]
Comment (jdrouet) [reply scripts/setup_integration/docker-compose.datadog-agent.yml:15]: I followed the datadog agent documentation 😓

File: Makefile [hunk 6]
```diff
@@ -368,6 +368,18 @@ ifeq ($(AUTODESPAWN), true)
 	@scripts/setup_integration_env.sh clickhouse stop
 endif
 
+.PHONY: test-integration-datadog-agent
+test-integration-datadog-agent: ## Runs Datadog Agent integration tests
+ifeq ($(AUTOSPAWN), true)
+	@scripts/setup_integration_env.sh datadog-agent stop
+	@scripts/setup_integration_env.sh datadog-agent start
+	sleep 30 # Many services are very slow... Give them a sec...
```
[the remark this answers is not in the document]
Comment (jdrouet) [reply Makefile:377]: but I have to adapt the test to wait for the service to be ready then?

**Review by jszwedko — APPROVED** [review]
Nice! I think we should unify this with your integration test refactoring in https://github.com/vectordotdev/vector/pull/10466 but otherwise looks good!

**Pull request merged**